### PR TITLE
Fixes #12885: Make scale-out (relay) server a plugin

### PR DESCRIPTION
**Pull request by fanf**
https://www.rudder-project.org/redmine/issues/12885

### CODE REVIEW

**Comment by fanf** [comment]
PR rebased

**Comment by fanf** [comment]
PR rebased

File: rudder-core/src/main/scala/com/normation/rudder/services/policies/write/AgentSpecificLogic.scala [hunk 1]
```diff
@@ -120,9 +120,11 @@ class AgentRegister {
...
    * If none is found, return a failure.
    */
-  def findMap[T](agentType: AgentType, osDetails: OsDetails)(f: AgentSpecificGeneration => Box[T]) = {
-    pipeline.find(handler => handler.handle(agentType, osDetails)) match {
-      case None    => Failure(s"We were unable to find how to create directive sequences for Agent type ${agentType.toString()} on '${osDetails.fullName}'. " +
+  def findMap[T](agentNodeProps: AgentNodeProperties)(f: AgentSpecificGeneration => Box[T]) = {
+    pipeline.find(handler => handler.handle(agentNodeProps)) match {
+      case None    =>
+        val relayMsg = if(agentNodeProps.isPolicyServer) { "(scale-out relay server)"} else ""
+        Failure(s"We were unable to find how to create directive sequences for Agent type ${agentNodeProps.agentType.toString()} on '${agentNodeProps.osDetails.fullName}' ${relayMsg}. " +
```
Comment (VinceMacBuche) [inline rudder-core/src/main/scala/com/normation/rudder/services/policies/write/AgentSpecificLogic.scala]: Maybe we could have a different message for policy server, clearer about the impact and what could be the cause (missing plugin ? ). I propose to change messages this way:

* non policy server:
We could not generate policies for node '${agentNodeProps.osDetails.fullName}' based on '${agentNodeProps.agentType.toString()}' agent. Maybe you are missing dedicated plugin ?

* policy server:
We could not generate policies for server '${agentNodeProps.osDetails.fullName}', therefore making updates for nodes behind it unavailable. Maybe you are missing "scale out" plugin?

**Comment by VinceMacBuche** [comment]
Apart a comment on the message this is ok!!

**Comment by fanf** [comment]
PR rebased

**Pull request merged**